nautobot/dcim/management/commands/migrate_location_contacts.py

```diff
@@ -0,0 +1,205 @@
+                self.stdout.write(f"    current contact phone: {location.contact_phone!r}")
+                self.stdout.write(f"    current contact email: {location.contact_email!r}")
+                self.stdout.write("")
+
+                # Output menu of choices of valid contacts/teams
+                for i, contact in enumerate(similar_contacts_and_teams, start=1):
+                    self.stdout.write(f"{self.style.WARNING(i)}: {contact._meta.model_name.title()}: {contact.name}")
+                    self.print_contact_fields(contact, rjust=len(str(i)) + len(contact._meta.model_name) + 2)
+                    self.stdout.write("")
+                self.stdout.write(self.style.WARNING("c") + ": Create a new Contact")
+                self.stdout.write(self.style.WARNING("t") + ": Create a new Team")
+                self.stdout.write(self.style.WARNING("s") + ": Skip this location")
+
+                # Retrieve desired contact/team from user input
+                while True:
+                    choice = input("Select a choice from the list of items: ")
+                    if choice == "s":
+                        self.stdout.write(f"Skipping location {location.display}")
+                        break
+                    elif choice == "c":
+                        selected_contact = self.create_new_contact_from_location(location, model=Contact)
+                        break
+                    elif choice == "t":
+                        selected_contact = self.create_new_contact_from_location(location, model=Team)
+                        break
+                    elif choice.isdigit() and 0 < int(choice) <= len(similar_contacts_and_teams):
+                        selected_contact = similar_contacts_and_teams[int(choice) - 1]
+                        break
+
+            if selected_contact is not None:
+                self.associate_contact_to_location(selected_contact, location)
+
+    def associate_contact_to_location(self, contact, location):
+        role, status = self.prompt_for_role_and_status()
+
+        # If email or phone fields are present in the location but not the contact, update the contact fields
+        updated_fields = {}
+        if location.contact_phone and not contact.phone:
+            contact.phone = location.contact_phone
+            updated_fields["phone"] = location.contact_phone
+        if location.contact_email and not contact.email:
+            contact.email = location.contact_email
+            updated_fields["email"] = location.contact_email
+        if updated_fields:
+            try:
+                contact.validated_save()
+                self.stdout.write(
+                    self.style.SUCCESS(
+                        f"Updated {contact._meta.model_name.title()} {contact.name} field(s): {updated_fields!r}"
+                    )
+                )
+            except ValidationError as e:
```

Comment:
Do we have known cases for this or is this purely defensive coding?

Reply:
I ran into it while developing this with my test data. The management command tried to create a contact identical to an existing contact and it failed due to uniqueness constraints.